.circleci/config.yml

```diff
@@ -12,44 +12,35 @@ commands:
   install_macos_dependencies:
     steps:
       - run: brew install protobuf cmake ccache libtool libspatialite pkg-config luajit curl wget czmq lz4 spatialite-tools unzip
-      - run: pip3 install requests shapely conan
+      - run: python3 -m pip install requests shapely conan
       - run: git clone https://github.com/kevinkreiser/prime_server --recurse-submodules && cd prime_server && ./autogen.sh && ./configure && make -j8 && make install
+  install_linux_dependencies:
+    steps:
+      - run: sudo bash ./scripts/install-linux-deps.sh
 
 jobs:
   build-docker-images:
-    docker:
-      - image: docker:18.06.3-ce-git
+    machine:
+      image: ubuntu-2004:2022.07.1
     steps:
-      - setup_remote_docker
       - checkout
       - run:
-          name: Build Valhalla Docker Images
+          name: Build and push Valhalla Docker image
           command: |
-            docker build --pull -f ./docker/Dockerfile-build --tag valhalla/valhalla:build-latest ./docker
             if [[ "${CIRCLE_BRANCH}" == "master" ]] || [[ ! -z "${CIRCLE_TAG}" ]]; then
-              docker build -f ./docker/Dockerfile-run --tag valhalla/valhalla:run-latest .
-            fi
-      - run:
-          name: Push Valhalla Docker Images
-          command: |
-            if [ "${CIRCLE_BRANCH}" == "master" ]; then
-              echo "$DOCKERHUB_PASS" | docker login -u "$DOCKERHUB_USERNAME" --password-stdin
-              docker push valhalla/valhalla:build-latest
-              docker push valhalla/valhalla:run-latest
-            elif [[ ! -z "${CIRCLE_TAG}" ]]; then
+              BUILD_TAG="${CIRCLE_TAG:-latest}"
```

Comment:
now we'd only build docker images on master and releases, but IMO that's fine, the other CI builds run on very similar systems